ddev/src/ddev/repo/core.py

```diff
@@ -69,56 +69,80 @@ def get(self, name: str) -> Integration:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_package:
                 yield integration
 
-    def iter_tiles(self) -> Iterable[Integration]:
+    def iter_tiles(self, selection: Iterable[str] = ()) -> Iterable[Integration]:
+        selected = self.__finalize_selection(selection)
         for path in sorted(self.repo.path.iterdir()):
             integration = self.__get_from_path(path)
-            if integration.is_tile:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_tile:
                 yield integration
 
-    def iter_testable(self) -> Iterable[Integration]:
+    def iter_testable(self, selection: Iterable[str] = ()) -> Iterable[Integration]:
+        selected = self.__finalize_selection(selection)
         for path in sorted(self.repo.path.iterdir()):
             integration = self.__get_from_path(path)
-            if integration.is_testable:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_testable:
                 yield integration
 
-    def iter_shippable(self) -> Iterable[Integration]:
+    def iter_shippable(self, selection: Iterable[str] = ()) -> Iterable[Integration]:
+        selected = self.__finalize_selection(selection)
         for path in sorted(self.repo.path.iterdir()):
             integration = self.__get_from_path(path)
-            if integration.is_shippable:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_shippable:
                 yield integration
 
-    def iter_agent_checks(self) -> Iterable[Integration]:
+    def iter_agent_checks(self, selection: Iterable[str] = ()) -> Iterable[Integration]:
+        selected = self.__finalize_selection(selection)
         for path in sorted(self.repo.path.iterdir()):
             integration = self.__get_from_path(path)
-            if integration.is_agent_check:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_agent_check:
                 yield integration
 
-    def iter_jmx_checks(self) -> Iterable[Integration]:
+    def iter_jmx_checks(self, selection: Iterable[str] = ()) -> Iterable[Integration]:
+        selected = self.__finalize_selection(selection)
         for path in sorted(self.repo.path.iterdir()):
             integration = self.__get_from_path(path)
-            if integration.is_jmx_check:
+            if selected and integration.name not in selected:
+                continue
+            elif integration.is_jmx_check:
                 yield integration
 
     def iter_changed(self) -> Iterable[Integration]:
-        changed_root_entries = {relative_path.split('/', 1)[0] for relative_path in self.repo.git.changed_files}
+        changed_root_entries = self.__get_changed_root_entries()
```

Comment:
Isn't `iter_changed` functionally equivalent to `iter_all` with an empty selection? I.e.

```python
    def iter_changed(self) -> Iterable[Integration]:
        return self.iter_all()
```

Which also makes `self.iter_all` look misnamed.